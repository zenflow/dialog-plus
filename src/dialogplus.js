import { DialogplusCore } from './dialogplus-core'
import { dialogplusPluginIcons } from './dialogplus-plugin-icons'

const Dialogplus = DialogplusCore.withPlugins(dialogplusPluginIcons)

export { DialogplusCore, dialogplusPluginIcons, Dialogplus }
