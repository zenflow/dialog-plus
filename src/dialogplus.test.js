/* eslint-env jest */

import { Dialogplus } from './dialogplus.js'

describe('Dialogplus', () => {
  it('Dialogplus.fire({content})', () => {
    const dialog = Dialogplus.create({ content: '<p>test</p>' })
    // Arggg so bad! need to `expect` *after* a wait on the micro tast queue
    return Promise.resolve().then(() => {
      expect(dialog.elements.container).toMatchSnapshot()
    })
  })
})
