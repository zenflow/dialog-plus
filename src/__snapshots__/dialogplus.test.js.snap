// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialogplus Dialogplus.fire({content}) 1`] = `
<div
  class="dialogplus--root"
>
  <div
    class="dialogplus--backdrop"
  />
  <div
    class="dialogplus--dialog"
  >
    <div
      class="dialogplus--content"
    >
      <p>
        test
      </p>
    </div>
  </div>
</div>
`;
